monkey/tests/unit_tests/monkey_island/cc/setup/test_server_setup.py

```diff
@@ -0,0 +1,90 @@
+from json import dumps
+from pathlib import Path
+
+import pytest
+
+import monkey_island.cc.setup.config_setup  # noqa: F401
+from monkey_island.cc.arg_parser import IslandCmdArgs
+from monkey_island.cc.server_setup import _extract_config
```

Comment:
Testing private/internal methods/functions is generally bad practice. Unit tests are for testing public interfaces.

Since `_extract_config()` is mostly just a wrapper around `get_server_config()`, let's just call that function instead. Then we can get rid of `test_malformed_json()` and simplify the call to `create_server_config()`.

Reply:
It's important to test what happens if the user provides a malformed JSON and it mostly uses the same fixtures to do so. I've tried splitting this up, but I don't think it's making this any more readable or maintainable, it just makes it more granular. I'm leaving it as it is.